packages/typescript/internal/ts_project.bzl

```diff
@@ -540,6 +546,27 @@ def ts_project_macro(
 
         args: List of strings of additional command-line arguments to pass to tsc.
 
+        transpiler: What tool to run that produces the JavaScript outputs.
+            By default, this is the string `tsc` which means to produce `.js` outputs
+            in the same action that does the type-checking to produce `.d.ts` outputs.
+            This is the simplest configuration, however `tsc` is slower than alternatives.
+            It also means developers must wait for the type-checking in the developer loop.
+
+            In theory, Persistent Workers (via the `supports_workers` attribute) remedies the
+            slow compilation time, however it adds additional complexity because the worker process
+            can only see one set of dependencies, and so it cannot be shared between different
+            `ts_project` rules. That attribute is documented as experimental, and may never graduate
+            to a better support contract.
+
+            Instead, you can pass a rule or macro that accepts these arguments:
+            `(name, srcs, js_outs, map_outs, args, data, tags, visibility)`
```

Comment:
Upon my experimentation using our own swc rules, I realized that this won’t work out of the box. macro does not accept many of these attributes. and some of these attributes get declared twice. 
https://github.com/aspect-build/rules_swc/blob/33d5a39030cb87b8d5bc2f03473429d49ab8657b/swc/swc.bzl#L38

so you end up having an error that you don’t know what to do with.

Reply:
Right, the swc_rule is what exposes this API